### DISCUSSION TITLE
Alternate way to show albums on artist page

### DISCUSSION THREAD

**Discussion by troycarpenter** [ampache/ampache#3314] [Ideas]
I didn't put this into a feature request yet, I wanted to bounce it off other first since there may be deep implementation implications.

Recently I was prompted to revisit my library tags by @mitchray, which led me to the wonderful (debatable) world of Musicbrainz.  I've known of Musicbrainz, but never really used it as a metadata source before.  As a result (and due to my off-mainstream music interests), I've already submitted new artists (and the albums I have) as well as albums missing from established artists.

In any case, due to artist collaborations, this has led to a situation where I have artists attributed to tracks on other artists albums where the first artist is just a guest, not the main artist.  In those cases, I now have the second artist's album in the album list of the first artist.  Quick example:
![image](https://user-images.githubusercontent.com/8419000/184169399-750d9e3e-15c9-49e7-8ac6-f97873cccd6c.png)
Here, the Quebe Sisters have appeared on a track on an album by Asleep at the Wheel and now Asleep at the Wheel's album appears as an album under the Quebe Sisters.

Now, depending on how you look at the Album designation, you can say they appeared on the album and it deserves to be there, or that the Album category should be for the listed artist.  I personally lean to the second interpretation.

Therefore, to start I would like to see another category listed on the web interface as "Appears On", which would be a list of albums (as defined by the release type) where the current artist appears on a track, but is not the album artist.  As such:

![image](https://user-images.githubusercontent.com/8419000/184173588-ed4009f4-8f66-41ab-bfe8-fc91160bca0b.png)

In that vein, I wouldn't mind seeing a separation of release type "Compilation" further based on album artist.  There is a difference between a Best Of compilation by a single artist, as opposed to a compilation by Various Artists.  However, that can easily be handled by tagging the release type as "Best Of" instead of compilation.

Other release types (EP, Soundtrack, or whatever the user defines) should be treated as is currently.

Since this is an interpretation of the data by the client, I don't think this changes anything on the clients.  For instance, it looks like Subsonic clients are simply sent every album on which an artist appears...if it shows up on the web client regardless of the displayed category, it shows up in a singular list of albums under the artist in the Subsonic Client (which has taken me a bit to get used to coming from a *sonic server that doesn't do that and made it easier to play all albums by a particular artist).

Thoughts?

**Comment by mitchray** [comment]
I agree they should appear separately where they aren't by the listed artist

In Ample I just show it as "{release_type} (appearance)" in those instances which keeps the release types grouped but I'm wondering now if it should just be a broader "Appears on" section like your example

![firefox_ccMzk3fcIY](https://user-images.githubusercontent.com/5735900/184495929-d937fc2c-e09b-4d10-a495-a0a9ab64481a.png)

**Reply by troycarpenter** [reply]
I do like the way Ample has them grouped.  For instance, the same artist I used above appears this way:
![image](https://user-images.githubusercontent.com/8419000/186215195-2dafd44b-c287-4edc-bb37-a5b664df79f4.png)
In this case, the albums where The Quebe Sisters only appear on a track (or two) but not as the album artists, it shows appearance.  Even for the holiday album (where I put "Holiday" in as the release type tag) it is grouped as I would expect.

**Comment by troycarpenter** [comment]
I also like this information (from a different artist in my library):
![image](https://user-images.githubusercontent.com/8419000/186219807-1e9e4309-e65c-4964-8c32-8b2270d97c33.png)
